datafusion/common/src/utils.rs

```diff
@@ -103,6 +111,53 @@ where
     Ok(low)
 }
 
+/// This function searches for a tuple of given values (`target`) among the given
+/// rows (`item_columns`) via a linear scan. It assumes that `item_columns` is sorted
+/// according to `sort_options` and returns the insertion index of `target`.
+/// Template argument `SIDE` being `true`/`false` means left/right insertion.
+pub fn linear_search<const SIDE: bool>(
```

Comment:
I didn't see `linear_search` used anywhere in this PR other than tests  -- I wonder if the intention was to call it in `datafusion/physical-expr/src/window/window_frame_state.rs` ? It looks like that currently has an inlined version of this function 🤔

Reply:
Yes, so the same logic has two drivers: One with a comparison function, one with `SortOptions`. We currently use the former, but also anticipate to use the latter in the near future (we plan a follow-up of this PR for GROUPS mode). As @mustafasrepo mentions, it also brings both search APIs in line, which is good too.

Reply:
My hesitation was that the tests are for `linear_search` -- the fact that you plan to use it in the future makes sense to me